### ISSUE TITLE
Segmentation fault on RaspberryPi Zero

### ISSUE THREAD

**Issue by parksj10** [rolandjitsu/grpc-cross-arm#1]
Hi,

I found your guide [here](https://rolandsdev.blog/cross-compile-grpc-for-arm-with-docker/) and it worked great all the way up to running the binaries on the Pi Zero... after which I got

`Segmentation fault`

I tried pulling down your most recent update and recompiling with the same effect. Am I missing something glaring me in the face? I've seen online that there are quite a few posts about grpc not working on Pi's, but it seems to be running ARMv6 (your compile doesn't seem to be version specific?)

Any help or guidance much appreciated

**Comment by parksj10** [comment]
leaving answer here in case someone comes looking... seems like [rPi Zero has an ARMv6Z](https://www.mbtechworks.com/hardware/raspberry-pi-model-comparison.html), which is not supported by the compiler used in this doc ([linux-gnueabihf supports armhf](https://packages.debian.org/stretch/gcc-arm-linux-gnueabihf), which requires your [ARM be v7 or higher...](https://superuser.com/a/1259737/1201625)).

TLDR

_./Dockerfile_
`ENV GNU_HOST=arm-linux-gnueabihf`
-->
`ENV GNU_HOST=arm-linux-gnueabi`

**Comment by rolandjitsu** [comment]
Hi @parksj10. Sorry for the late reply, but my wife and I just had a 👶 and I suddently don't have time to do anything. Your suggestion makes sense, just use a different compiler (maybe the one in [raspberrypi/tools](https://github.com/raspberrypi/tools)). You may also find more info [here](https://stackoverflow.com/a/35132679/1092007).
